uniffi_bindgen/src/bindings/kotlin/templates/CallbackInterfaceTemplate.kt

```diff
@@ -0,0 +1,75 @@
+{% let type_name = obj.name()|class_name_kt %}
+public interface {{ type_name }} {
+    {% for meth in obj.methods() -%}
+    fun {{ meth.name()|fn_name_kt }}({% call kt::arg_list_decl(meth) %})
+    {%- match meth.return_type() -%}
+    {%- when Some with (return_type) %}: {{ return_type|type_kt -}}
+    {%- else -%}
+    {%- endmatch %}
+    {% endfor %}
+}
+
+{% let canonical_type_name = format!("CallbackInterface{}", type_name) %}
+{% let callback_internals = format!("{}Internals", canonical_type_name) -%}
+{% let callback_interface_impl = format!("{}FFI", canonical_type_name) -%}
+
+internal class {{ callback_interface_impl }} : ForeignCallback {
+    @Suppress("TooGenericExceptionCaught")
+    override fun invoke(handle: Long, method: Int, args: RustBuffer.ByValue): RustBuffer.ByValue {
+        return {{ callback_internals }}.handleMap.callWithResult(handle) { cb -> 
+            when (method) {
+                0 -> {{ callback_internals }}.drop(handle)
+                {% for meth in obj.methods() -%}
+                {% let method_name = format!("invoke_{}", meth.name())|fn_name_kt -%}
+                {{ loop.index }} -> this.{{ method_name }}(cb, args)
+                {% endfor %}
+                else -> RustBuffer.ByValue()
```

Comment:
In some future iteration, I guess this is where we'd signal an error back to the Rust code. Any thoughts on how that might work? (Or should we just forbid it for now?)

Reply:
Broadly agreed, the more symmetry we can have between `interface` and `callback interface` the better things are likely to be.

Reply:
Could you please add a brief `TODO` comment in the code here with a link to the follow-up issue? I think it's important that we remind our future selves that the `else` branch here is a temporary situation.